test/headers.js

```diff
@@ -142,12 +142,22 @@ test('remove null value headers', async t => {
 	t.false(Reflect.has(headers, 'user-agent'));
 });
 
-test('remove undefined value headers', async t => {
+test('setting a header to undefined keeps the old value', async t => {
 	const {body} = await got(s.url, {
 		headers: {
 			foo: undefined
```

Comment:
Shouldn't this be `'user-agent'`?

Reply:
Yeah, must missed that when was merging with master. Fixed.